### DISCUSSION TITLE
Loading data in an SSR'ed component

### DISCUSSION THREAD

**Discussion by eecue** [sveltejs/kit#9204] [Q&A] [answered]
I have a component that injects a photo gallery based on a keyword string in a blog post.  It works fine when I use onMount but I want to have this pre-rendered/SSR as my whole site is static.  I can't figure out how to make it work.  I saw this thread but I wasn't able to get it working for my use case:

https://stackoverflow.com/questions/67422216/sveltekit-load-not-called-from-component-but-works-as-a-page

**Comment by teemingc** [comment, marked as answer]
Unfortunately, I don't think there is data loading at the component level. It is only for pages and layouts.
See the following regarding data loading https://kit.svelte.dev/docs/load

**Reply by eecue** [reply]
That's unfortunate.

**Comment by itssumitrai** [comment]
Its actually better for page performance to fetch all the data in parallel for your page using the load functions. Slow /non-important requests can be streamed https://svelte.dev/blog/streaming-snapshots-sveltekit

**Reply by eecue** [reply]
All my pages are pre-rendered static html as noted above.
